src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax;
+
+public class Main {
+    private static FigureSupplier figureSupplier = new FigureSupplier();
+
+    public static void main(String[] args) {
+        Figure[] figures = new Figure[6];
+        for (int i = 0; i < figures.length; i++) {
+            if (i < 3) {
```

Comment:
```suggestion
            if (i < figures.length / 2) {
```
3 - magic number, but better use expression